server/apps/immich/src/api-v1/album/album-repository.ts

```diff
@@ -283,4 +284,17 @@ export class AlbumRepository implements IAlbumRepository {
 
     return this.albumRepository.save(album);
   }
+
+  async getSharedAlbumCount(userId: string, assetId: string): Promise<number> {
```

Comment:
is this "the number of albums where asset=assetId and that are shared with user=userId"?
If so, the name `getSharedAlbumCount` might be misleading. Maybe something like `getSharedWithUserAlbumCount` (not perfect, naming things is  hard problem). Alternatively, add a doc comment to the method so it's easy to grasp what count this returns.